src/main/java/org/ehcache/sizeof/ObjectGraphWalker.java

```diff
@@ -194,26 +193,58 @@ long walk(VisitorListener visitorListener, Object... root) {
      * @param refClass the type
      * @return A collection of fields to be visited
      */
-    private Collection<Field> getFilteredFields(Class<?> refClass) {
-        SoftReference<Collection<Field>> ref = fieldCache.get(refClass);
-        Collection<Field> fieldList = ref != null ? ref.get() : null;
+    private Collection<Function<Object, Object>> getFilteredFields(Class<?> refClass) {
+        SoftReference<Collection<Function<Object, Object>>> ref = fieldCache.get(refClass);
+        Collection<Function<Object, Object>> fieldList = ref != null ? ref.get() : null;
         if (fieldList != null) {
             return fieldList;
         } else {
-            Collection<Field> result;
-            result = sizeOfFilter.filterFields(refClass, getAllFields(refClass));
+            Collection<Field> fields = sizeOfFilter.filterFields(refClass, getAllFields(refClass));
+            Collection<Function<Object, Object>> accessors = new ArrayList<>(fields.size());
             if (USE_VERBOSE_DEBUG_LOGGING && LOG.isDebugEnabled()) {
-                for (Field field : result) {
+                for (Field field : fields) {
                     if (Modifier.isTransient(field.getModifiers())) {
                         LOG.debug("SizeOf engine walking transient field '{}' of class {}", field.getName(), refClass.getName());
                     }
                 }
             }
-            fieldCache.put(refClass, new SoftReference<>(result));
-            return result;
+
+            for (Field field : fields) {
+                try {
+                    accessors.add(readUsingReflection(field));
+                } catch (SecurityException e) {
+                    LOG.error("Security settings prevent Ehcache from accessing the subgraph beneath '{}'" +
+                            " - cache sizes may be underestimated as a result", field, e);
+                    accessors.add(readUsingUnsafe(field));
+                } catch (RuntimeException e) {
+                    LOG.warn("The JVM is preventing Ehcache from accessing the subgraph beneath '{}'" +
+                            " - cache sizes may be underestimated as a result", field, e);
```

Comment:
I test this PR and found the logs are full of this warning. I think we should only log this if unsafe fails. Otherwise, it's confusing since we are actually not understimating the size.

Reply:
@chrisdennis Could you please address this? I tested the latest version in Impala. The logs are still full of such warnings:
```
W1012 12:40:35.496349  2656 ObjectGraphWalker.java:220] 4a48697e1f87e06d:102ed89f00000000] The JVM is preventing Ehcache from accessing the subgraph beneath 'final jdk.internal.loader.URLClassPath jdk.internal.loader.ClassLoaders$AppClassLoader.ucp' - cache sizes may be underestimated as a result
Java exception follows:
java.lang.reflect.InaccessibleObjectException: Unable to make field final jdk.internal.loader.URLClassPath jdk.internal.loader.ClassLoaders$AppClassLoader.ucp accessible: module java.base does not "opens jdk.internal.loader" to unnamed module @10823d72 
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:340)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:280)
        at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:176)
        at java.base/java.lang.reflect.Field.setAccessible(Field.java:170)
        at org.ehcache.sizeof.ObjectGraphWalker.readUsingReflection(ObjectGraphWalker.java:232)
        at org.ehcache.sizeof.ObjectGraphWalker.getFilteredFields(ObjectGraphWalker.java:214)
        at org.ehcache.sizeof.ObjectGraphWalker.walk(ObjectGraphWalker.java:162)
        at org.ehcache.sizeof.SizeOf.deepSizeOf(SizeOf.java:83)
        at org.apache.impala.catalog.local.CatalogdMetaProvider$SizeOfWeigher.weigh(CatalogdMetaProvider.java:2014)
        at com.google.common.cache.LocalCache$Segment.setValue(LocalCache.java:2010)
        at com.google.common.cache.LocalCache$Segment.replace(LocalCache.java:2956)
        at com.google.common.cache.LocalCache.replace(LocalCache.java:4258)
        at org.apache.impala.catalog.local.CatalogdMetaProvider.loadWithCaching(CatalogdMetaProvider.java:542)
        at org.apache.impala.catalog.local.CatalogdMetaProvider.loadIcebergApiTable(CatalogdMetaProvider.java:1058)
        at org.apache.impala.catalog.local.LocalIcebergTable.loadIcebergTableViaMetaProvider(LocalIcebergTable.java:90)
        at org.apache.impala.catalog.local.LocalTable.load(LocalTable.java:116)
        at org.apache.impala.catalog.local.LocalDb.getTable(LocalDb.java:127)
        at org.apache.impala.analysis.StmtMetadataLoader.getMissingTables(StmtMetadataLoader.java:310)
        at org.apache.impala.analysis.StmtMetadataLoader.loadTables(StmtMetadataLoader.java:165)
        at org.apache.impala.analysis.StmtMetadataLoader.loadTables(StmtMetadataLoader.java:141)
        at org.apache.impala.service.Frontend.doCreateExecRequest(Frontend.java:2014)
        at org.apache.impala.service.Frontend.getTExecRequest(Frontend.java:1926)
        at org.apache.impala.service.Frontend.createExecRequest(Frontend.java:1748)
        at org.apache.impala.service.JniFrontend.createExecRequest(JniFrontend.java:164)
```

Can we simply remove this log since the following accessor got from readUsingUnsafe() will work?